src/argilla/client/feedback/training/frameworks/trl.py

```diff
@@ -0,0 +1,383 @@
+            )
+            self.trainer_kwargs["config"] = PPOConfig()
+            self.training_args_kwargs["reward_model"] = None
+            self.training_args_kwargs["length_sampler_kwargs"] = {"min_value": 1, "max_value": 10}
+            self.training_args_kwargs["generation_kwargs"] = {
+                "min_length": -1,
+                "top_k": 0.0,
+                "top_p": 1.0,
+                "do_sample": True,
+            }
+        else:
+            self.training_args_kwargs["evaluation_strategy"] = "no" if self._eval_dataset is None else "epoch"
+            self.training_args_kwargs["logging_steps"] = 30
+            self.training_args_kwargs["logging_steps"] = 1
+            self.training_args_kwargs["num_train_epochs"] = 1
+
+    def init_model(self, new: bool = False) -> None:
+        """
+        Initializes a model.
+        """
+        from transformers import (
+            AutoModelForCausalLM,
+            AutoModelForSequenceClassification,
+            AutoTokenizer,
+            PreTrainedModel,
+            PreTrainedTokenizer,
+        )
+        from trl import AutoModelForCausalLMWithValueHead, create_reference_model
+
+        if isinstance(self._task, (TrainingTaskForSFT, TrainingTaskForDPO)):
+            auto_model_class = AutoModelForCausalLM
+        elif isinstance(self._task, TrainingTaskForPPO):
+            auto_model_class = AutoModelForCausalLMWithValueHead
+        elif isinstance(self._task, TrainingTaskForRM):
+            auto_model_class = AutoModelForSequenceClassification
+
+        self._transformers_model: PreTrainedModel = auto_model_class.from_pretrained(self._model)
+        self._transformers_tokenizer: PreTrainedTokenizer = AutoTokenizer.from_pretrained(self._model)
+        self._transformers_tokenizer.pad_token = self._transformers_tokenizer.eos_token
+        self._transformers_model.config.pad_token_id = self._transformers_tokenizer.pad_token_id
+
+        if isinstance(self._task, (TrainingTaskForPPO, TrainingTaskForDPO)):
+            self._transformers_ref_model: PreTrainedModel = create_reference_model(self._transformers_model)
+        if new:
+            self._transformers_model.to(self.device)
+
+    def update_config(self, **kwargs) -> None:
+        """
+        Updates the configuration of the trainer, but the parameters depend on the trainer.subclass.
+        """
+        from transformers import TrainingArguments
+
+        self.training_args_kwargs.update(filter_allowed_args(TrainingArguments.__init__, **kwargs))
+        self.training_args_kwargs.update(filter_allowed_args(PPOArgs.__init__, **kwargs))
+        self.trainer_kwargs.update(filter_allowed_args(self.trainer_cls.__init__, **kwargs))
+
+    def predict(self, text: Union[List[str], str], as_argilla_records: bool = True, **kwargs) -> None:
```

Comment:
maybe we should allow for generative predictions?

Reply:
Then I think a more logical next step is to add a separate generate method?

Reply:
I'll make an issue for this